### PR TITLE
[exporter/azuremonitorexporter] fix: correct parent for span events

### PR DESCRIPTION
**Pull request by alaendle**
**Description:** Corrected parent for exported span events.
For details please take a look at the corresponding issue.
I think the change is pretty straight forward - one line added.
However I'm not a go-expert - so there might exist a more elegant implementation.

**Link to tracking Issue:** #27233

**Testing:** Adapted existing units test.

**Documentation:** n/a

(fyi @pcwiese)

### CODE REVIEW

File: exporter/azuremonitorexporter/trace_to_envelope.go
Comment (rajkumar-rangaraj) [inline exporter/azuremonitorexporter/trace_to_envelope.go:136]: `newEnvelope` method has the entry for `envelope.Tags[contracts.OperationParentId]`, that seems to be correct. 
`envelope.Tags[contracts.OperationParentId] = traceutil.SpanIDToHexOrEmptyString(span.ParentSpanID())`


> spanEventEnvelope.Tags[contracts.OperationParentId] = traceutil.SpanIDToHexOrEmptyString(span.SpanID())

Setting the current span ID as the parent does not seem correct. Did you validate with an app that issue #27233 is resolved by this update?

Providing the .NET implementation here for reference - https://github.com/Azure/azure-sdk-for-net/blob/2d1996d14d430c541c5beabad612b98a01fb4a4e/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs#L19

Java Implementation - https://github.com/Azure/azure-sdk-for-java/blob/9e6bab9bb50d865ed8b4e5ab3198e3fc56d9e468/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java#L288

Comment (alaendle) [reply exporter/azuremonitorexporter/trace_to_envelope.go:136]: To be honest I haven't validated the change in conjunction with AppInsights. However the result looks correct:
Before the change:
grafik
Now:
grafik

Comment (alaendle) [reply exporter/azuremonitorexporter/trace_to_envelope.go:136]: @rajkumar-rangaraj - the nifty difference here - we are not mapping spans, we are mapping logs - e.g. compare with                 https://github.com/Azure/azure-sdk-for-java/blob/9e6bab9bb50d865ed8b4e5ab3198e3fc56d9e468/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java#L186 and lines around.

Comment (rajkumar-rangaraj) [reply exporter/azuremonitorexporter/trace_to_envelope.go:136]: Yes, this is correct in the context of logs. All logs will have the parent ID labeled as 'spanid.' However, you are modifying the trace envelope, which corresponds to spans. Logs are handled differently; therefore, the change should be applied to the log exporter file. The current proposed change might break the UI in other scenarios.

Comment (rajkumar-rangaraj) [reply exporter/azuremonitorexporter/trace_to_envelope.go:136]: Moving your changes to `log_to_envelope.go` should resolve an issue.

Comment (alaendle) [reply exporter/azuremonitorexporter/trace_to_envelope.go:136]: @rajkumar-rangaraj - first huge sorry - with the `LogDataMapper`I've pointed you to the wrong direction. We are still and only talking about trace data. However I've found the more correct line that underpins my change also in the `azure-sdk-for-java` - https://github.com/Azure/azure-sdk-for-java/blob/9e6bab9bb50d865ed8b4e5ab3198e3fc56d9e468/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java#L779 
For `Event`s the parent span is the span of the current operation. That also makes sense, because the events are just annotations of the current operation/tracing span. So maybe we should take one step back, and find out how far we already agree.
1. There is a problem with the original implementation (#27233) - the original behaviour could and should be improved?
2. The provided PR solves this issue - and has no unintended side effects?
3. Is the form of the changes okay, or should the patch be rephrased?

Comment (rajkumar-rangaraj) [reply exporter/azuremonitorexporter/trace_to_envelope.go:136]: Got it. It makes sense for span events.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by rajkumar-rangaraj** [comment]
@pcwiese, Could you please review this PR.

**Comment by rajkumar-rangaraj** [comment]
> [CLA Not Signed](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/9522512/196414933/30659/#/?version=2)
> 
> * [❌](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/9522512/196414933/30659/#/?version=2) - login: @alaendle / name: Andreas Ländle . The commit ([6c5a730](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/6c5a730fb89ab0ec03eb1a217f6004d0c1912de5)) is not authorized under a signed CLA. [Please click here to be authorized](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/9522512/196414933/30659/#/?version=2). For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

@alaendle Could you please fix this one?

**Review by pcwiese — APPROVED** [review]

**Comment by rajkumar-rangaraj** [comment]
@open-telemetry/collector-contrib-approvers Could you please approve the pipeline and help merge this PR. It has got an approval from the code owner.

**Review by songy23 — COMMENTED** [review]
Please add a changelog

**Review by songy23 — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Comment by alaendle** [comment]
Pinging @djaglowski - maybe you could approve/merge this PR? Or is still something missing?

**Comment by alaendle** [comment]
This is becoming more and more of a frustrating experience. The PR is now 3 month old and I still can't predict whether or when he will make it to the finish line.

**Comment by bryan-aguilar** [comment]
Apologies @alaendle. I understand how this can be frustrating. I see we have codeowner and approver review already. I went ahead and stuck the `ready to merge` label which should give better visibility to the maintainers. Please just keep an eye out on the CI to make sure it stays green. You shouldn't need to continually update from main unless a merge conflict pops up.

**Review by bryan-aguilar — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**
